packages/patternfly-4/react-core/src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import styles from '@patternfly/patternfly/components/FormControl/form-control.css';
+import { css } from '@patternfly/react-styles';
+import { Omit } from '../../helpers/typeUtils';
+import { FormEvent } from 'react';
+
+export enum inputTypes {
+  text = 'text',
+  date = 'date',
+  datetimeLocal = 'datetime-local',
+  email = 'email',
+  month = 'month',
+  number = 'number',
+  password = 'password',
+  search = 'search',
+  tel = 'tel',
+  time = 'time',
+  url = 'url'
+}
+
+export interface TextInputProps extends Omit<React.HTMLProps<HTMLInputElement>, 'onChange'> {
+  /** Additional classes added to the TextInput. */
+  className?: string;
+  /** Flag to show if the input is disabled. */
+  isDisabled?: boolean;
+  /** Flag to show if the input is read only. */
+  isReadOnly?: boolean;
+  /** Flag to show if the input is required. */
+  isRequired?: boolean;
+  /** Flag to show if the input is valid or invalid. */
+  isValid?: boolean;
+  /** A callback for when the input value changes. */
+  onChange?: (value: string, event: React.FormEvent<HTMLInputElement>) => void;
+  /** Type that the input accepts. */
+  type?: 'text' | 'date' | 'datetime-local' | 'email' | 'month' | 'number' | 'password' | 'search' | 'tel' | 'time' | 'url';
+  /** Value of the input. */
+  value?: string | number;
+  /** Aria-label. The input requires an associated id or aria-label. */
+  'aria-label'?: string;
```

Comment:
I believe we still want this to default to null for now.  We have a future story to change these to undefined.